cli/src/pcluster/validators/cluster_validators.py

```diff
@@ -747,6 +749,24 @@ def _validate(self, mount_dir: str):
             )
 
 
+class DeletionPolicyValidator(Validator):
+    """Print warning message when deletion policy is set to Delete or Retain."""
+
+    def _validate(self, deletion_policy: str, name: str):
+        if deletion_policy == DELETE_POLICY:
+            self._add_failure(
+                f"The DeletionPolicy is set to {DELETE_POLICY}. The storage '{name}' will be deleted when you remove "
```

Comment:
Minor: I would slightly rephrase to

```
The storage '{name}' will be deleted when you remove it from the configuration and perform a cluster update or a cluster delete.
```

Similarly to the below message for Retain